app/controllers/reg/SummaryController.scala

```diff
@@ -85,7 +85,17 @@ trait SummaryController extends AuthenticatedController with CommonService with
                   case AccountingDetailsSuccessResponse(response) => response
                   case _ => throw new Exception("could not find company accounting details")
                 }
-                Ok(Summary(details.companyName, details.jurisdiction, accountDates, ppobAddress, rOAddress, ctContactDetails, tradingDetails, takeoverDetails, cc))
+                optTakeoverDetails match {
+                  case Some(takeoverDetails) =>
+                    if (takeoverDetails.replacingAnotherBusiness && (takeoverDetails.businessName.isEmpty || takeoverDetails.businessTakeoverAddress.isEmpty || takeoverDetails.previousOwnersAddress.isEmpty || takeoverDetails.previousOwnersName.isEmpty)) {
```

Comment:
```suggestion
                  case Some(TakeoverDetails(replacingAnotherBusiness = true, Some(_), Some(_), Some(_), Some(_)) => Redirect(controllers.takeovers.routes.TakeoverInformationNeededController.show())
```
This should simplify your logic here, I'm not 100% that TakeoverDetails is the correct model name so it'd be worth checking that.